src/main/host/syscall/fcntl.rs

```diff
@@ -127,10 +127,10 @@ fn fcntl(ctx: &mut ThreadContext, args: &SysCallArgs) -> SyscallResult {
             if let PosixFile::Pipe(pipe) = desc.get_file() {
                 SysCallReg::from(i32::try_from(pipe.borrow().max_size()).unwrap())
             } else {
-                Err(Errno::EINVAL)?
+                return Err(Errno::EINVAL.into());
```

Comment:
I'm not sure this transformation is a win, but I guess I'm willing to along with clippy :)

Reply:
I wasn't sure originally, but I think I agree with clippy now. The explicit `return` shows that we intend to always return an error, and imo makes it easier to skip over that branch when looking through the code.